tests/PHPStan/Rules/Comparison/data/bug-9475.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Bug9475;
+
+use Stringable;
+
+class Foo
+{
+	private string $foo = 'Foo';
+
+	/** @param mixed[] $array */
+	function mustReport(Bug9475 $foo, array $array): void
+	{
+		echo $this->foo;
+		echo $this->$foo;
+		echo $this->$array;
+		echo $this->$this;
+	}
+}
+
+class Bar implements Stringable
+{
+	private string $foo = 'Foo';
+
+	function mustReport(Bug9475 $foo): void
+	{
+		echo $this->foo;
+		echo $this->$foo;
+		echo $this->$this;
```

Comment:
`$this` is a Stringable so this line is legal, but heuristically it looks like a typo.
It is also possible to alert another message for this code.